src/Core/Component.js

```diff
@@ -465,5 +470,44 @@ function enqueueStateCallback (element, owner, callback) {
 			return callback.call(owner, owner.state, owner.props, owner.context)
 	} catch (err) {
 		invokeErrorBoundary(element, err, SharedSiteCallback)
+	} 
+}
+
+/* istanbul ignore next */
+/**
+ * @param {function} type Component class (constructor) or SFC function
+ * @param {object} props Object of current component props
+ * @returns {object|null} Object of validation functions per prop name,
+ * 							or null if no propTypes are declared for the component
+ */
+function getPropTypes(type, props) {
+	var propTypes = type[SharedPropTypes]
+
+	if (typeof propTypes === 'function')
+		propTypes = propTypes(props)
+
+	return propTypes && typeof propTypes === 'object'
+		? propTypes
+		: null
+}
+
+/* istanbul ignore next */
+/**
+ * @param {function} type Component class (constructor) or (render) function
+ * @param {object} props Object of current component props
+ * @throws {any} will be thrown on validation error
+ */
+function checkPropTypes(type, props) {
+	var propTypes = getPropTypes(type, props)
+
+	if (propTypes) {
+		var componentName = getDisplayName(type)
+
+		for (var propName in propTypes) {
+			var error = propTypes[propName](props, propName, componentName, 'prop', null)
+
+			if (error)
```

Comment:
could add a check for `Error` type for enforcing correctness

Reply:
@thysultan @Zolmeister 
Hey guys, what exactly do you want now? .. as you gave me different task descriptions regarding the error check ;-)

Option 1: if (error instanceof Error) // => as given in the first example snippet by @thysultan 
Option 2: if (error && typeof error.message === 'string') // => Quote: "Error-like to support cross realm instances"
Option 3: if (error) // => Quote:  "maybe this is enough(consider any truthy value as cause for exception)"

I personally prefer Option 1 as I cannot think of any use case where the "cross realm capability" regarding that error object is needed ... but maybe that's just due to lack of imagination ;-)

Reply:
This is what I meant:

```js
if (error) {
    if (error instanceof Error) {
        throw error
    else {
        throw new Error 'Return value must be of type `Error`'
    }
}
```